packages/view-util/lib/dimensionIdentifiers.js

```diff
@@ -0,0 +1,53 @@
+import * as ns from '@view-builder/core/ns.js'
+import { schema } from '@tpluscode/rdf-ns-builders'
+import TermMap from '@rdfjs/term-map'
+
+const expandedProperties = new TermMap([
+  [schema.name, 'lang'],
+  [schema.termCode, 'code'],
+])
+
+export async function populateDimensionIdentifiers(pointer, metaLookup) {
+  const cubes = pointer
+    .out(ns.view.dimension)
+    .out(ns.view.from)
+    .out(ns.view.source)
+    .out(ns.view.cube)
+    .terms
+
+  const cubeKeys = await metaLookup.getCubeKeys(...cubes)
+  const dimensionIdentifiers = await metaLookup.getDimensionIdentifiers()
+
+  pointer
+    .out(ns.view.dimension)
+    .forEach((dimension) => {
+      const from = dimension.out(ns.view.from)
+      const property = from.out(ns.view.path).term
+      const cube = from.out(ns.view.source).out(ns.view.cube).terms.shift()
+
+      if (!property) {
+        return
+      }
+
+      if (!cube) {
+        // maybe a lookup source
+        const joinedProperty = from.out(ns.view.join).out(ns.view.from).out(ns.view.path).term
+        const { identifier } = dimensionIdentifiers.get(joinedProperty) || {}
+        const suffix = expandedProperties.get(property) || '?'
+
+        dimension.addOut(schema.identifier, `${identifier}_${suffix}`)
```

Comment:
I would add some documentation somewhere explaining the business logic of the identifiers. 

It can be derived from the code; however, I think adding a short documentation somewhere wouldn't hurt

Reply:
Indeed. There is also an example csv https://github.com/StatistikStadtZuerich/APD/issues/155#issuecomment-1315069482